petastorm/hdfs/namenode.py

```diff
@@ -247,7 +249,13 @@ def hdfs_connect_namenode(cls, url, driver='libhdfs3'):
         :param driver: An optional driver identifier
         :return: Pyarrow HDFS connection object.
         """
-        return pyarrow.hdfs.connect(url.hostname or 'default', url.port or 8020, driver=driver)
+        if LooseVersion(pyarrow.__version__) < LooseVersion('0.12.0'):
+            hostname = url.hostname or 'default'
+            driver = driver
+        else:
+            hostname = six.text_type(url.hostname or 'default')
```

Comment:
Not sure I understand the 'default' thing

Reply:
Added a comment:
```python
# According to pyarrow.hdfs.connect:
#    host : NameNode. Set to "default" for fs.defaultFS from core-site.xml
# So we pass 'default' as a host name if the url does not specify one (i.e. hdfs:///...)
```